src/LightSaml/SymfonyBridgeBundle/Bridge/Store/Request/RequestStateSessionStore.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace LightSaml\SymfonyBridgeBundle\Bridge\Store\Request;
+
+use LightSaml\Store\Request\AbstractRequestStateArrayStore;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class RequestStateSessionStore extends AbstractRequestStateArrayStore
+{
+    protected RequestStack $requestStack;
+    protected string $providerId;
+    protected string $prefix;
+
+    public function __construct(
+        RequestStack $requestStack,
+        string $providerId,
+        string $prefix = 'saml_request_state_'
+    ) {
+
+        $this->requestStack = $requestStack;
+        $this->providerId = $providerId;
+        $this->prefix = $prefix;
+    }
+
+    protected function getKey(): string
+    {
+        return sprintf('%s_%s', $this->providerId, $this->prefix);
```

Comment:
Maybe we could store this `sprintf()` in a `protected string $key` attribute, since `providerId` and `prefix` are used only in this operation and you call `getKey` in both `getArray` and `setArray`?

Reply:
done